web3c/index.js

```diff
@@ -20,8 +20,11 @@ let rejectWeb3 = () => {};
  *
  * @param {Object} provider is an object conforming to the web3 provider interface.
  * @param {Object?} web3 is an optional web3 object to override the localWeb3.
+ * @param {String} options is a set of options containing configurations to web3c.
+ *                 Currently the only key is keyManagerPublicKey, an optional hex
+ *                 string to configure the remote key manager for signature validation.
  */
-module.exports = function (provider, web3) {
+module.exports = function (provider, web3, options) {
   if (web3) {
```

Comment:
can we simplify the localWeb3 logic a bit now that it can be passed in as a constructor arg?
now that we have several backing services (storage, MraeBox implementation, web3 implementation), i wonder if it makes sense to have all of those wrapped in a 'state' type object we pass around through the library?

Reply:
Agreed lets throw those into one object.

Reply:
Lets allow the key manager storage to be passed in through options as well. Then we can simultaneously address https://github.com/oasislabs/web3c.js/issues/137

Reply:
I gave this a shot. Let me know if you have any ideas w.r.t. simplifying the localWeb3 logic.